### ISSUE TITLE
[Q] How to dump the current session to an image?

### ISSUE THREAD

**Issue by NightMachinary** [JuliaLang/PackageCompiler.jl#591]
This image will simply be a serialization of the current environment, with all its loaded packages, defined variables, etc. 

Like Common Lisp's save-lisp-and-die.

**Comment by IanButterworth** [comment]
> Someone interested in this could refactor this to allow to ccall it and output a module immidiately.

I tried the minimum idea but this hits either a segfault during `jl_create_system_image` or `Task cannot be serialized`
https://github.com/IanButterworth/julia/commit/ba3e049f2e140f7f06d904173d91bc194819da3f

Evidently more is needed, but I'm sharing here in case it's useful (I don't think it's advanced-enough to open even a WIP PR to base)

**Comment by KristofferC** [comment]
> Task cannot be serialized

Yes, I think there can be no running tasks which makes it tricky since e.g. the REPL is running. So you might have to somehow exit the REPL before saving the sysimage and then starting it back up after it is saved, or something along those lines...

**Comment by IanButterworth** [comment]
Initially I was just trying to do it at exit, setting JLOptions after start then calling exit, but I think the way `__init__`s are handled might be incompatible with that approach
